code/modules/hydroponics/grown/grass_carpet.dm

```diff
@@ -36,17 +36,7 @@
 			continue
 		grassAmt += 1 + round(G.seed.potency * tile_coefficient)
 		qdel(G)
-	var/obj/item/stack/tile/GT = new stacktype(user.loc)
-	while(grassAmt > GT.max_amount)
-		GT.amount = GT.max_amount
-		grassAmt -= GT.max_amount
-		GT = new stacktype(user.loc)
-	GT.amount = grassAmt
-	for(var/obj/item/stack/tile/T in user.loc)
-		if((T.type == stacktype) && (T.amount < T.max_amount))
-			GT.merge(T)
-			if(GT.amount <= 0)
-				break
+	new stacktype(user.drop_location(), grassAmt)
 	qdel(src)
 	return
```

Comment:
You don't need a trailing return here, might as well remove it as you fix the other thing.